framework/di/Container.php

```diff
@@ -527,7 +531,17 @@ public function resolveCallableDependencies(callable $callback, $params = [])
                 } elseif (isset(Yii::$app) && Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
                     $args[] = $obj;
                 } else {
-                    $args[] = $this->get($className);
+                    // If the argument is optional we catch not instantiable exceptions.
+                    try {
+                        $args[] = $this->get($className);
+                    } catch (NotInstantiableException $e) {
```

Comment:
Should not it be just check, if default value is specified, then use default value avoiding instantiating anything?